src/components/DpDataTable/DpDataTable.vue

```diff
@@ -597,31 +649,23 @@ export default {
      * Therefore, we have a normal table, get the auto-sized cell width and set the layout to fixed afterwards.
      */
     if (this.isResizable || this.isTruncatable) {
-      const firstRow = this.tableEl.firstChild
-      const tableHeaders = Array.prototype.slice.call(firstRow.childNodes)
-      tableHeaders.forEach(tableHeader => {
-        /**
-         * Some of childNodes of the first table row are not Element nodes but comments or text.
-         * This originates in the Vue template compiler leaving empty html comments when rendering
-         * falsy `v-if` blocks. We allow only nodeType "Element" to access its `getBoundingClientRect` api.
-         */
-        if(tableHeader.nodeType === 1) {
-          const width = tableHeader.getBoundingClientRect().width
-          tableHeader.style.width = width + 'px'
-        }
-      })
+      const firstRow = this.tableEl.getElementsByTagName('tr')[0]
+      const tableHeaders = firstRow ? firstRow.children : null
+
+      this.setColsWidth(tableHeaders)
 
       this.tableEl.style.tableLayout = 'fixed'
       this.tableEl.classList.add('is-fixed')
 
       // Remove styles set by initialMaxWidth and initialWidth after copying rendered width into th styles
       if (this.isResizable) {
-        const tableRows = Array.from(this.tableEl.children[1].children)
-        tableRows.forEach(tableRow => {
-          Array.from(tableRow.children).forEach(cell => {
-            cell.firstChild.style.width = null
-            cell.firstChild.style.maxWidth = null
-            cell.firstChild.style.minWidth = null
+        Array.from(tableHeaders).forEach(tableRow => {
```

Comment:
I don't understand what is achieved here. `tableHeaders` holds a dom nodeList of all `th` elements, so (with the above naming tweak) it could read...

```suggestion
        Array.from(tableHeaderElement).forEach(th => {
```

So within the outer forEach, the `th` variable should hold a dom node element, not an array. Why are you applying a nested loop here?

Reply:
yes, you are right. I did not change here the logic that was applied before. Thanks for finding this issue. done here https://github.com/demos-europe/demosplan-ui/pull/835/commits/19afd6dc9b016b75745ba7a70f1425433ee8a3d7